x-pack/legacy/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_panel.tsx

```diff
@@ -41,152 +50,185 @@ export interface OperationFieldSupportMatrix {
   fieldByOperation: Partial<Record<OperationType, string[]>>;
 }
 
-export const IndexPatternDimensionPanelComponent = function IndexPatternDimensionPanel(
-  props: IndexPatternDimensionPanelProps
-) {
+type Props = Pick<
+  DatasourceDimensionDropProps<IndexPatternPrivateState>,
+  'layerId' | 'columnId' | 'state' | 'filterOperations'
+>;
+const getOperationFieldSupportMatrix = /* _.memoize( */ (
```

Comment:
Commented code. Don't forget to remove before merging.

Reply:
Left a TODO here because I can't find a way to memoize this safely.